grails-app/assets/javascripts/spApp/service/toolsService.js

```diff
@@ -1,430 +1,470 @@
-                    $http.post(url, m, _httpDescription('executeRemote')).then(function (response) {
-                        uiScope.externalTaskId = response.data.id;
-                        uiScope.statusUrl = LayersService.url() + '/tasks/status/' + response.data.id;
-
-                        // Create log entry for an external tool when the task is created with the taskId
-                        LoggerService.log('Tool', uiScope.toolName, '{ "taskId": "' + uiScope.externalTaskId + '"}')
-
-                        $timeout(function () {
-                            _checkStatus(uiScope);
-                        }, 5000)
-                    });
-
-                    return $q.when(false)
+  'use strict';
+  /**
+   * @memberof spApp
+   * @ngdoc service
+   * @name ToolsService
+   * @description
+   *   List of spatial-hub tools. Includes client side tools with names matching Tool*Service. Includes
+   * tools from spatial-service/capabilities.
+   */
+  angular.module('tools-service', [])
+  .factory('ToolsService',
+      ['$injector', '$q', '$http', '$timeout', 'MapService', 'LayersService',
+        'LoggerService', 'LayoutService',
+        function ($injector, $q, $http, $timeout, MapService, LayersService,
+            LoggerService, LayoutService) {
+          var cap = $SH.layersServiceCapabilities;
+          var viewConfig = $SH.viewConfig;
+          var localToolServices = {};
+          var promises = [];
+
+          initLocalTools();
+
+          var _httpDescription = function (method, httpconfig) {
+            if (httpconfig === undefined) {
+              httpconfig = {};
+            }
+            httpconfig.service = 'ToolsService';
+            httpconfig.method = method;
+
+            return httpconfig;
+          };
+
+          /*
+          uiScope is ToolCtrl
+           */
+          function executeRemote(uiScope, inputs) {
+            var m = {};
+            m['input'] = inputs;
+            m['name'] = uiScope.toolName;
+
+            var url = $SH.baseUrl + '/portal/postTask?sessionId='
+                + $SH.sessionId;
+            $http.post(url, m, _httpDescription('executeRemote')).then(
+                function (response) {
+                  uiScope.externalTaskId = response.data.id;
+                  uiScope.statusUrl = LayersService.url() + '/tasks/status/'
+                      + response.data.id;
+
+                  // Create log entry for an external tool when the task is created with the taskId
+                  LoggerService.log('Tool', uiScope.toolName,
+                      '{ "taskId": "' + uiScope.externalTaskId + '"}')
+
+                  $timeout(function () {
+                    _checkStatus(uiScope);
+                  }, 5000)
+                },
+                function (error) {
+                    var message = '';
+                  if(error.headers()['content-type'].indexOf('json')>0){
```

Comment:
indentation is wrong (`if` is to left of line above it) - did you mix spaces and tabs? Use IDE to format the code.

Reply:
Reverted and add codes in